File: Assignment - 4/Q-1.py
import numpy as np
import array as arr

class stack:
    def __init__(self):
        self.arr = []
        self.max = 0
        self.top = 0
    
    def push(self, num):
#        if self.isFull:
#            print('Stack is Full.')
        print(self.top)
        self.arr[self.top] = num
        self.top += 1

    def pop(self):
        pass

    def print(self):
        if isEmpty:
            print('Stack is empty.')
        
        for i in range(self.len()):
            print(self.arr[i], end=" ")
        print(" ")


    def isFull(self):
        return false

    def isEmpty(self):
        return false

    def peek(self):
        print(self.arr[self.top-1])
    
    def len(self):
        return self.top


if __name__=='__main__':
    choice = None
    s = stack()

    len   = int(input('Enter the length of the stack : '))
    s.max = len

    while True: 
        print('\n1) Push the element.')
        print('2) Pop the element.')
        print('3) Check if stack is empty or not.')
        print('4) Check if stack is full or not.')
        print('5) Print the stack.')
        print('6) Print top element.')
        print('Press Q to Exit.')
        
        choice = input("\nPlease Enter Your Choice : ")

        if choice == '1':
            num = int(input('Enter the number to insert into stack : '))
            s.push(num)

        elif choice == '2':
            s.pop()

        elif choice == '3':
            s.print()

        elif choice == '4':
            s.isFull()

        elif choice == '5':
            s.isEmpty()

        elif choice == '6':
            s.peek()

        else:
            break

File: Assignment - 4/Q-1.cpp
/*

1) Develop a Menu driven program to demonstrate the following operations of
Arrays ——MENU——- 1.CREATE
2.DISPLAY
3.INSERT
4.DELETE
5.SEARCH
6.EXIT

*/

#include <iostream>
using namespace std;

#define len 1000

int top=-1;
int arr[len];

void push(int num)
		{
			top += 1;
			arr[top] = num;
		}

		void pop()
		{
			top -= 1;
		}

		void print()
		{
			cout<<top<<endl;
			for(int i=0; i<=top; i++)
			{
				cout<<arr[i]<<"\t";
			}
			cout<<endl;
		}

		void peek()
		{
			cout<<arr[top]<<endl;
		}

		bool isFull()
		{
			return (top == len);
		}

int main()
{	
	int choice = 0, exit = 1, element=0;

	cout<<"\nEnter the length of stack : ";
	//cin>>len;
    
    do{
    	cout<<"--MENU--"<<endl;
		cout<<"1. PUSH\n2. POP\n3. DISPLAY\n4. isFULL\n5. PEEK\n6. EXIT\n\nPlease Enter your Choice : ";
 		cin>>choice;

 		switch(choice)
 		{
	  	case 1:
		  	 cout<<"Enter a number : ";
			 cin>>element;
			 push(element);
 			 break;
		case 2:pop();
			  break; 
        case 3:print();
			  break;
        case 4:
              if (isFull)
			  {
			  	cout<<"True"<<endl;
			  }
			  else
			  {
				  cout<<"False"<<endl;
			  }
			  
			  break;
        case 5:peek();
			  break;
        case 6:	exit=0;
			  break;
        }
	}while(exit);
 	
	return 0;
}
